zerver/views/registration.py

```diff
@@ -180,7 +181,7 @@ def accounts_register(request):
             except KeyError:
                 pass
         form = RegistrationForm(postdata, realm_creation=realm_creation)
-        if not password_auth_enabled(realm):
+        if not password_auth_enabled(realm) or is_remote:
             form['password'].field.required = False
```

Comment:
So one thing I don't understand about this piece of code is that I am not aware of cases when the field is required.  Like, the default in `forms.py` is also False.  But maybe this code is correct and the default in `forms.py` should be `True`?